modules/standard/Random.chpl

```diff
@@ -576,20 +733,38 @@ module Random {
       /*
         Returns the next value in the random stream.
 
-        Reals are between 0 and 1; integers are in the range of the integer.
-        :returns: The next value in the random stream as type :type:`eltType`.
+        Generated reals are in [0,1] - both 0.0 and 1.0 are
+        possible values.
+
+        Generated integers cover the full value range of the integer.
+
+        :arg resultType: the type of the result. Defaults to :type:`eltType`.
+          `resultType` must be the same or a smaller size number.
+        :returns: The next value in the random stream as type :type:`resultType`.
        */
-      proc getNext(): eltType {
+      proc getNext(type resultType=eltType): resultType {
```

Comment:
I'm not convinced that permitting getNext() to take an optional resultType is a good idea.  It seems suspicious to me to say "this stream generates a (proven) good series of values of type t" and then to say "but you can pull out values of type t' as well" (and expect that to mean anything in the general case).